src/driver/i2c.c

```diff
@@ -16,28 +16,41 @@
 
 pthread_mutex_t i2c_mutex;
 
-void iic_init() {
-    pthread_mutex_init(&i2c_mutex, NULL);
-}
+#define IIC_PORTS 4
 
-static char *iic_ports[IIC_PORTS] = {
+static char *IIC_DEVS[IIC_PORTS] = {
     "/dev/i2c-0",
     "/dev/i2c-1",
     "/dev/i2c-2",
     "/dev/i2c-3",
 };
 
-static int iic_open(char *port) {
-    // int i2c_fd = open ("/dev/i2c-0", O_RDONLY);
-    int i2c_fd = open(port, O_RDONLY);
-    if (i2c_fd < 0) {
-        LOGE("iic_open %s failed [1=Right,2=Main,3=Left]", port);
+int g_iic_fds[IIC_PORTS];
+static int g_iic_report_once[IIC_PORTS] = {0};
+
+int iic_is_port_ready(int port) {
```

Comment:
`stdbool.h` has been standardized for while :D lets please go with a bool.

Reply:
Will do!